packages/cfd/src/features/hooks/useIsEuRegion.ts

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react';
+import { useLandingCompany } from '@deriv/api';
+
+const useIsEuRegion = () => {
+    const { data: landing_company, ...rest } = useLandingCompany();
+
+    const isEuRegion = useMemo(() => {
+        if (!landing_company) return false;
+
+        const euShortcodeRegex = /^(maltainvest|malta|iom)$/;
+        const euExcludedRegex = /^mt$/;
+
+        const { gaming_company, financial_company, mt_gaming_company, mt_all_company, id: residence } = landing_company;
+        const financialShortcode = financial_company?.shortcode;
+        const gamingShortcode = gaming_company?.shortcode;
+        const mtGamingShortcode = mt_gaming_company?.financial?.shortcode || mt_all_company?.swap_free?.shortcode;
+        const isCurrentMf = financial_company?.shortcode === 'maltainvest';
+
+        return (
+            isCurrentMf ||
+            (financialShortcode || gamingShortcode || mtGamingShortcode
+                ? (euShortcodeRegex.test(financialShortcode as unknown as string) && gamingShortcode !== 'svg') ||
+                  euShortcodeRegex.test(gamingShortcode as string)
+                : euExcludedRegex.test(residence as string))
+        );
```

Comment:
Pls make a variable on the top and return for better readability

Reply:
```JavaScript
    const isEuRegion = useMemo(() => {
        if (!landing_company) return false;

        const euShortcodeRegex = /^(maltainvest|malta|iom)$/;
        const euExcludedRegex = /^mt$/;

        const { gaming_company, financial_company, mt_gaming_company, mt_all_company, id: residence } = landing_company;
        const financialShortcode = financial_company?.shortcode; // mf:maltainvest cr:svg
        const gamingShortcode = gaming_company?.shortcode; // mf:undefined cr:svg
        const mtGamingShortcode = mt_gaming_company?.financial?.shortcode || mt_all_company?.swap_free?.shortcode; // mf:undefined cr:svg
        const isCurrentMf = financial_company?.shortcode === 'maltainvest'; // mf:true cr:false

        const shortcodes = financialShortcode || gamingShortcode || mtGamingShortcode;
        const isFinancialEuAndNotGamingSvg =
            euShortcodeRegex.test(financialShortcode as unknown as string) && gamingShortcode !== 'svg';
        const isGamingEu = euShortcodeRegex.test(gamingShortcode as string);
        const isResidenceEu = euExcludedRegex.test(residence as string);

        return isCurrentMf || shortcodes ? isFinancialEuAndNotGamingSvg || isGamingEu : isResidenceEu;
    }, [landing_company]);
```

can advice if is this any better? @hamza-deriv @thisyahlen-deriv 🙏🏼

Reply:
Use one casing for the variables probaly snake_case.
```
const isEuRegion = useMemo(() => {
    if (!landing_company) return false;

    const eu_shortcode_regex = /^(maltainvest|malta|iom)$/;
    const eu_excluded_regex = /^mt$/;

    const { gaming_company, financial_company, mt_gaming_company, mt_all_company, id: residence } = landing_company;
    const financial_shortcode = financial_company?.shortcode;
    const gaming_shortcode = gaming_company?.shortcode; 
    const mt_gaming_shortcode = mt_gaming_company?.financial?.shortcode || mt_all_company?.swap_free?.shortcode; 
    const is_current_mf = financial_company?.shortcode === 'maltainvest'; 

    const shortcodes = financial_shortcode || gaming_shortcode || mt_gaming_shortcode;
    const is_financial_eu_and_not_gaming_svg =
        euShortcodeRegex.test(financialShortcode as unknown as string) && gamingShortcode !== 'svg';
    const is_gaming_eu = eu_shortcode_regex.test(gaming_shortcode as string);
    const is_residence_Eu = eu_excluded_regex.test(residence as string);

    const is_eu_based_on_shortcodes = shortcodes && (is_financial_eu_and_not_gaming_svg || is_gaming_eu);
    const is_eu_based_on_residence = !shortcodes && is_residence_eu;

    return is_current_mf || is_eu_based_on_shortcodes || is_eu_based_on_residence;
}, [landing_company]);

```